docs/metric_deletion_and_retention.md

```diff
@@ -0,0 +1,133 @@
+```
+
+## Deletion of metric
+
+Promscale allows you to delete an entire metric both via SQL and API.
+
+### SQL
+
+You can delete an entire metric from the database using the function `prom_api.drop_metric(metric_name_to_be_dropped text)`.
+
+**Example:**
+
+In order to delete the metric `container_cpu_load_average_10s`
+
+```postgresql
+SELECT prom_api.drop_metric('container_cpu_load_average_10s');
+```
+
+### API
+
+In order to delete a metric via API, you can use the `/delete_series` API and pass the metric name to be deleted, as the
+matcher. However, please note that, for deleting data in Promscale, you will need to enable permissions for advanced users. This is done by setting
+[`web-enable-admin-api`](https://github.com/timescale/promscale/blob/master/docs/cli.md#general-flags) flag to `true`.
+
+URL query parameters:
+
+* **match[]=<series_selector>**: Repeated label matcher argument that selects the series to delete. At least one match[] argument must be provided.
+
+```
+POST /delete_series
+PUT /delete_series
+```
+
+**Example:**
+
+In order to delete the metric `container_cpu_load_average_10s` using the `/delete_series` API
+
+```shell
+curl -X POST -g http://localhost:9201/delete_series?match[]=container_cpu_load_average_10s
+```
+
+## Deletion of series/metric by time (SQL)
+
+Deletion of series or metrics based on time can be done only through SQL. The support for time-based deletion in the
+`/delete_series` endpoint will be added in near future.
+
+### SQL
+
+Since data inside hypertable can be present in form of compressed and uncompressed chunks, we need to first decompress
+all the compressed chunks and then perform any time-based deletion. We may recompress the chunks after deletion if required.
+
+**Example:**
+
+In order to delete all data from `container_cpu_load_average_10s` metric that is beyond 10 hours of time,
+
+Lets decompress all chunks related to the hypertable of the given metric.
+
+```postgresql
+SELECT decompress_chunks(show_chunks('prom_data.container_cpu_load_average_10s'));
+```
+
+Then, we perform the deletion query.
+
+```postgresql
+DELETE FROM prom_data.container_cpu_load_average_10s WHERE time > Now() - interval '10 hour';
+```
+
+Now, we can recompress the remaining data.
+
+```postgresql
+SELECT compress_chunks(show_chunks('prom_data.container_cpu_load_average_10s'));
+```
+
+## Metric Retention
+
+Promscale offers full control over data retentions i.e., you can set a default data retention period as well as overwrite the default on a per-metric basis.
+
+SQL command to set the default retention policy to two days for all metrics that do not have custom retention policy already specified.
+
+```
+SELECT prom_api.set_default_retention_period(INTERVAL '1 day' * 2)
+```
+
+SQL command to set a custom retention policy for a specific metric.
+
+```
+SELECT prom_api.set_metric_retention_period(container_cpu_usage_seconds_total, INTERVAL '1 day' * 2)
```

Comment:
Add a retention interval to something new, The custom retention policy is the same as the default in this example.